### ISSUE TITLE
Backtrace functionality is specific to glibc, build fails on some standard libraries

### ISSUE THREAD

**Issue by nivardus** [h2o/h2o#373]
`src/main.c` fails to build when using [musl][musl] or any C standard library that does not implement **glibc's** `execinfo.h`.

```sh
/.../h2o/src/main.c:51:22: fatal error: execinfo.h: No such file or directory
 #include <execinfo.h>
                      ^  
```

`execinfo.h` is not part of the C standard library, but was added to [provide backtraces][backtrace] in **h2o's** logs.

[musl]: http://www.musl-libc.org/
[backtrace]: https://github.com/h2o/h2o/commit/95b4d4e48cea094bd70e306b768fe8f0a76bbe8d

**Comment by JillTyrell** [comment]
You'll need to change all instances of __linux__ to __GLIBC__ in main.c, then it will build successfully against musl.

**Comment by nivardus** [comment]
Thanks, @JillTyrell --addressed in PR #374

**Comment by kazuho** [comment]
@nivardus @JillTyrell Thank you for your help!  Tha changes have been merged to master.
